extensions/functions_string.yaml

```diff
@@ -88,7 +88,11 @@ scalar_functions:
       from the beginning of the string to begin starting to search for pattern matches can be
       specified using the `position` argument. Specifying `1` means to search for matches
       starting at the first character of the input string, `2` means the second character, and so
-      on. The `position` argument should be a positive non-zero integer.
+      on. The `position` argument should be a positive non-zero integer. The regular
```

Comment:
Not your fault but why does `position` use "1-based" indices?

Reply:
SQL systems tend to use 1-based indices for positions in strings, [as explained here](https://dba.stackexchange.com/questions/125950/why-do-string-functions-on-sql-based-database-servers-start-at-position-1-instea). Looking at all the other functions already defined in [functions_string.yaml](https://github.com/substrait-io/substrait/blob/main/extensions/functions_string.yaml), it looks like we've chosen to follow this convention. I'm not sure whether this was an explicit choice, but regardless, it's probably beyond the scope of this PR to discuss it.